<?xml version="1.0" ?>
<robot name="caddy_bot" xmlns:xacro="http://ros.org/wiki/xacro">

  <link name="base_footprint"/>

  <joint name="base_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
    <origin xyz="0.0 0.0 0.132" rpy="0 0 0"/>
  </joint>
  <link name="base_link"/>

  <joint name="wheel_left_joint" type="fixed">
    <parent link="base_link"/>
    <child link="wheel_left_link"/>
    <origin xyz="0.0 0.285 0.0" rpy="-1.5708 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="wheel_left_link"/>

  <joint name="wheel_right_joint" type="fixed">
    <parent link="base_link"/>
    <child link="wheel_right_link"/>
    <origin xyz="0.0 -0.285 0.0" rpy="-1.5708 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="wheel_right_link"/>

  <joint name="caster_back_joint" type="fixed">
    <parent link="base_link"/>
    <child link="caster_back_link"/>
    <origin xyz="-0.2175 0 0.12" rpy="-1.5708 0 0"/>
  </joint>
  <link name="caster_back_link"/>

  <joint name="imu_joint" type="fixed">
    <parent link="base_link"/>
    <child link="imu_e2box_link"/>
    <origin xyz="0.028 0.04 0.05" rpy="0 0 1.5708"/>
  </joint>
  <link name="imu_e2box_link"/>

  <joint name="camera_joint" type="fixed">
    <parent link="base_link"/>
    <child link="camera_link"/>
    <origin xyz="0.15 0 0.31" rpy="0 0 0"/>
  </joint>
  <link name="camera_link"/>
</robot>